src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,46 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Track;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type.equals(Bulldozer.class)) {
+            List<Bulldozer> bulldozers = new BulldozerProducer().get();
+            return new ArrayList<>(bulldozers);
+        }
+        if (type.equals(Excavator.class)) {
+            List<Excavator> excavators = new ExcavatorProducer().get();
+            return new ArrayList<>(excavators);
+        }
+        if (type.equals(Track.class)) {
+            List<Track> tracks = new TrackProducer().get();
+            return new ArrayList<>(tracks);
+        }
+        throw new NullPointerException("type can't be null");
```

Comment:
better return some other type of exception, it can be IllegalArgumentException

Reply:
done